ZenPacks/zenoss/Import4/bin/imp4util.py

```diff
@@ -0,0 +1,259 @@
+                log.debug('found %s ...' % _rs[0])
+                break
+    return _list
+
+
+# list all services excluding those names containing the substrings in the list
+def _list_svcs_ex(exlist):
+    log.info('Listing services')
+
+    _list = []
+    _output = subprocess.check_output(_svc_cmd() + " list", shell=True).split('\n')
+
+    for _line in _output:
+        _rs = re.sub("[\s,]", " ", _line).split()
+        if (len(_rs) < 2 or
+            _rs[1].lower() == 'serviceid'):
+            log.debug('Skipping %s ...' % _line)
+            continue
+
+        _inc = True
+        for _ss in exlist:
+            if _rs[0].lower().find(_ss) >= 0:
+                _inc = False
+                break
+        if _inc:
+            log.debug('found %s ...' % _rs[0])
+            _list.append(_rs[1])
+
+    return _list
+
+
+def _tokens(_line):
+    return re.sub("[\s,]", " ", _line).split()
+
+
+def _is_stopped(slist):
+    for svc in slist:
+        _line = subprocess.check_output(_svc_cmd() + " status %s | grep %s" % (svc, svc),
+                                        shell=True)
+        _rs = _tokens(_line)
+        log.debug(_rs)
+
+        # ignore wrong services
+        if len(_rs) < 3:
+            continue
+
+        # ignore parents
+        if _rs[2] == '0':
+            continue
+
+        if _rs[2] != 'Stopped':
+            log.debug('%s not stopped ...' % _rs[0])
+            return False
+    return True
+
+
+def _is_running(slist):
+    for svc in slist:
+        _line = subprocess.check_output(_svc_cmd() + " status %s | grep %s" % (svc, svc),
+                                        shell=True)
+        _rs = _tokens(_line)
+        # ignore wrong services
+        if len(_rs) < 3:
+            continue
+
+        # ignore parents
+        if _rs[2] == '0':
+            continue
+
+        if _rs[2] != 'Running':
+            log.debug('%s not running ...' % _rs[0])
+            return False
+    return True
+
+
+def _stop_svc(id):
+    log.info('Stopping %s ...' % id)
+    subprocess.call(_svc_cmd() + " stop %s" % id, shell=True)
+
+
+def _start_svc(svc_name):
+    log.info('Staring %s ...' % svc_name)
+    subprocess.call(_svc_cmd() + " start %s" % svc_name, shell=True)
+    pass
+
+
+# stop all services except the root service, Imp4Mariadb, and mariadb*
+def stop_services(args):
+    log.info('Stopping services')
+
+    slist = _list_svcs_ex(['mariadb', 'zenoss.resmgr', 'opentsdb', 'hbase', 'hmaster', 'regionserver', 'zookeeper'])
+    for svc in slist:
+        _stop_svc(svc)
+
+    _time = 0
+    while (_time < 600 and
+           not _is_stopped(slist)):
+        log.debug('Waiting services to stop...')
+        _time += 10
+        time.sleep(10)
+
+    if _time >= 600:
+        log.error('Service stop failed ...')
+        sys.exit(1)
+
+
+# start all services in the control center
+def start_all_services(args):
+    log.info('Start services')
+
+    slist = _list_svcs_ex([])
+    for svc in slist:
```

Comment:
Yes, you are correct. To separate this performance improvement, I'll make the change in next iteration. Thanks.